File: src/backend/fitConnect/fitConnect/src/main/java/com/fitconnet/dto/entities/UserDTO.java
package com.fitconnet.dto.entities;

import java.util.Set;

import com.fitconnet.enums.Role;

import lombok.Data;

/**
 * Data Transfer Object (DTO) representing a user.
 */
@Data
public class UserDTO {
	/**
	 * The first name of the user.
	 */
	private String firstName;

	/**
	 * The last name of the user.
	 */
	private String lastName;

	/**
	 * The username of the user.
	 */
	private String userName;

	/**
	 * The age of the user.
	 */
	private Integer age;

	/**
	 * The email address of the user.
	 */
	private String email;

	/**
	 * The password of the user.
	 */
	private String password;

	/**
	 * The roles associated with the user.
	 */
	private Set<Role> roles;
}

File: src/src-frontend/src/model/userDTO.js
class UserDTO {
    constructor(id, firstName, lastName, username, age, email, password, image, roles) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.username = username;
        this.age = age;
        this.email = email;
        this.password = password;
        this.image = image;
        this.roles = roles;
        this.createdActivities = [];
        this.invitedActivities = [];
        this.friends = [];
        this.notifications = [];
        this.comments = [];
    }
}